examples/eSealing/src/lib.rs

```diff
@@ -0,0 +1,444 @@
+) -> InitResult<State<S>> {
+    Ok(State::new(state_builder))
+}
+
+/// Register a new file.
+///
+/// It rejects if:
+/// - It fails to parse the parameter.
+/// - If file hash has already been registered by the same witness
+///   (sender_account).
+/// - If a smart contract tries to register the file hash.
+#[receive(
+    contract = "eSealing",
+    name = "registerFile",
+    parameter = "String",
+    error = "ContractError",
+    mutable,
+    enable_logger
+)]
+fn register_file<S: HasStateApi>(
+    ctx: &impl HasReceiveContext,
+    host: &mut impl HasHost<State<S>, StateApiType = S>,
+    logger: &mut impl HasLogger,
+) -> Result<(), ContractError> {
+    // Ensure that only accounts can register a file
+    let sender_account = match ctx.sender() {
+        Address::Contract(_) => bail!(ContractError::OnlyAccount),
+        Address::Account(account_address) => account_address,
+    };
+
+    let file_hash: String = ctx.parameter_cursor().get()?;
+
+    let (state, builder) = host.state_and_builder();
+
+    // Ensure that every account can only register each file hash once.
+    ensure!(!state.witness_exists(&file_hash, sender_account), ContractError::AlreadyRecorded);
+
+    let slot_time = ctx.metadata().slot_time();
+
+    let exists = state.file_exists(&file_hash);
+
+    match exists {
+        true => {
+            // File exists => Add record
+            state.add_record(&file_hash, sender_account, slot_time);
+            logger.log(&Event::WitnessAdded(WitnessAddedEvent {
+                file_hash,
+                witness: sender_account,
+                timestamp: slot_time,
+            }))?;
+        }
+        false => {
+            // File does not exist => Add file and add record
+            state.add_file(&file_hash, builder);
+            logger.log(&Event::FileAdded(FileAddedEvent {
+                file_hash: file_hash.clone(),
+            }))?;
+
+            state.add_record(&file_hash, sender_account, slot_time);
+            logger.log(&Event::WitnessAdded(WitnessAddedEvent {
+                file_hash,
+                witness: sender_account,
+                timestamp: slot_time,
+            }))?;
+        }
+    }
+
+    Result::Ok(())
+}
+
+/// Get all records about a specific file hash.
+///
+/// It rejects if:
+/// - It fails to parse the parameter.
+#[receive(
+    contract = "eSealing",
+    name = "getFile",
+    parameter = "String",
+    error = "ContractError",
+    return_value = "(Vec<AccountAddress>, Vec<Timestamp>)"
+)]
+fn get_file<S: HasStateApi>(
+    ctx: &impl HasReceiveContext,
+    host: &impl HasHost<State<S>, StateApiType = S>,
+) -> ReceiveResult<(Vec<AccountAddress>, Vec<Timestamp>)> {
+    let file_hash: String = ctx.parameter_cursor().get()?;
+    let state = host.state();
+    let exists = state.file_exists(&file_hash);
+
+    match exists {
+        true => state.get_records(&file_hash),
+        false => Result::Ok(([].to_vec(), [].to_vec())),
+    }
```

Comment:
```suggestion
    state.get_records(&file_hash)
```
It looks to me like `get_records` already returns an empty list if the hash does not exist.